cluster/photon-controller/setup-prereq.sh

```diff
@@ -0,0 +1,238 @@
+#
+# It uses the names for the tenant, project, and flavors as specified in the
+# config-common.sh file
+#
+# If you want to do this by hand, this script is equivalent to the following
+# Photon Controller commands (assuming you haven't edited config-common.sh
+# to change the names)
+#
+# photon target set https://192.0.2.2
+# photon tenant create kube-tenant
+# photon tenant set kube-tenant
+# photon resource-ticket create --tenant kube-tenant --name kube-resources --limits "vm.memory 1000 GB, vm 1000 COUNT"
+# photon project create --tenant kube-tenant --resource-ticket kube-resources --name kube-project --limits "vm.memory 1000 GB, vm 1000 COUNT"
+# photon project set kube-project
+# photon -n flavor create --name "kube-vm" --kind "vm" --cost "vm 1 COUNT, vm.cpu 1 COUNT, vm.memory 2 GB"
+# photon -n flavor create --name "kube-disk" --kind "ephemeral-disk" --cost "ephemeral-disk 1 COUNT"
+# photon image create kube.vmdk -n kube-image -i EAGER
+#
+# Note that the kube.vmdk can be downloaded as specified in the documentation.
+
+KUBE_ROOT=$(dirname "${BASH_SOURCE[0]}")/../..
+# shellcheck source=./util.sh
+source "${KUBE_ROOT}/cluster/photon-controller/util.sh"
+
+function main {
+  verify-cmd-in-path photon
+  set-target
+  create-tenant
+  create-project
+  create-vm-flavor "${PHOTON_MASTER_FLAVOR}" "${SETUP_MASTER_FLAVOR_SPEC}"
+  if [ "${PHOTON_MASTER_FLAVOR}" != "${PHOTON_NODE_FLAVOR}" ]; then
+    create-vm-flavor "${PHOTON_NODE_FLAVOR}" "${SETUP_NODE_FLAVOR_SPEC}"
+  fi
+  create-disk-flavor
+  create-image
+}
+
+function parse-cmd-line {
+  PHOTON_TARGET=${1:-""}
+  PHOTON_VMDK=${2:-""}
+
+  if [[ "${PHOTON_TARGET}" = "" || "${PHOTON_VMDK}" = "" ]]; then
+    echo "Usage: setup-prereq <photon target> <path-to-kube-vmdk>"
+    echo "Target should be a URL like https://192.0.2.1"
+    echo ""
+    echo "This will create the following, based on the configuration in config-common.sh"
+    echo "    * A tenant named ${PHOTON_TENANT}"
+    echo "    * A project named ${PHOTON_PROJECT}"
+    echo "    * A VM flavor named ${PHOTON_MASTER_FLAVOR}"
+    echo "    * A disk flavor named ${PHOTON_DISK_FLAVOR}"
+    echo "It will also upload the Kube VMDK"
+    echo ""
+    echo "It creates the tenant with a resource ticket (quota) that may"
+    echo "be inappropriate for your environment. For a production"
+    echo "environment, you should configure these to match your"
+    echo "environment."
+    exit 1
+  fi
+
+  echo "Photon Target: ${PHOTON_TARGET}"
+  echo "Photon VMDK: ${PHOTON_VMDK}"
+}
+
+function set-target {
+  ${PHOTON} target set "${PHOTON_TARGET}" > /dev/null 2>&1
+}
+
+function create-tenant {
+  local rc=0
+  local output
+
+  ${PHOTON} tenant list | grep "\t${PHOTON_TENANT}$" > /dev/null 2>&1 || rc=$?
+  if [[ ${rc} -eq 0 ]]; then
+    echo "Tenant ${PHOTON_TENANT} already made, skipping"
+  else
+    echo "Making tenant ${PHOTON_TENANT}"
+    rc=0
+    output=$(${PHOTON} tenant create "${PHOTON_TENANT}" 2>&1) || rc=$?
```

Comment:
Previously you referred me to the Google bash style guide which shows the style I used. If you think it's important to switch, I will. 

https://google.github.io/styleguide/shell.xml?showone=Checking_Return_Values#Checking_Return_Values